posthog/temporal/data_imports/pipelines/pipeline/utils.py

```diff
@@ -353,11 +355,17 @@ def append_partition_key_to_table(
             elif mode == "datetime":
                 key = normalized_partition_keys[0]
                 date = row[key]
+
+                if partition_format == "day":
+                    date_format = "%Y-%m-%d"
+                else:
+                    date_format = "%Y-%m"
```

Comment:
So, my biggest problem with this is that, in theory, we are now going to end up with a lot of partitions. With a single years worth of data, we are jumping from 12 partitions to 365, and it gets worse as time goes by.

This leads me to think this change is more of a short term patch that is opening us up for trouble in the future.

Reply:
[Discussed on Slack here](https://posthog.slack.com/archives/C019RAX2XBN/p1744796167648889?thread_ts=1744793157.395779&cid=C019RAX2XBN)